[docs] Make it clear ray[all] is not recommended (#49837)
ray-project/ray: doc/source/ray-overview/installation.rst

          * - `pip install -U "ray[rllib]"`
            - Core, Tune, RLlib
          * - `pip install -U "ray[all]"`

>             - Core, Dashboard, Cluster Launcher, Data, Train, Tune, Serve, RLlib (not recommended, specify the extras you need like shown below instead)

Reviewer: ```suggestion
            - Core, Dashboard, Cluster Launcher, Data, Train, Tune, Serve, RLlib. This option isn't recommended. Specify the extras you need as shown below instead.
```
